registry/storage/driver/hdfsweb/fs.go

```diff
@@ -0,0 +1,452 @@
+// Refer to github.com/vladimirvivien/gowfs
+
+package hdfsweb
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/url"
+	"strconv"
+	"time"
+)
+
+const (
+	webHdfsVer string = "/webhdfs/v1"
+)
+
+// FileSystem provides the implementation of sending http requests to hdfs server,
+// and the parameters to build the http url.
+type FileSystem struct {
+	NameNodeHost string
+	NameNodePort string
+	Username     string
+	BlockSize    uint64
+	BufferSize   int
+	Replication  uint16
+}
+
+// Create creates a new file and stores its content in HDFS.
+func (fs *FileSystem) Create(
+	reader io.Reader,
+	path string) error {
+
+	transport := http.Transport{Dial: dialTimeout}
+	client := http.Client{
+		Transport: &transport,
+	}
+
+	v := url.Values{
+		"user.name":   []string{fs.Username},
+		"op":          []string{"CREATE"},
+		"overwrite":   []string{"true"},
+		"blocksize":   []string{strconv.FormatInt(int64(fs.BlockSize), 10)},
+		"replication": []string{strconv.FormatInt(int64(fs.Replication), 10)},
+		"permission":  []string{"0666"},
+		"buffersize":  []string{strconv.FormatInt(int64(fs.BufferSize), 10)},
+	}
+
+	u, err := fs.buildRequestURL(path, v)
+	if err != nil {
+		return err
+	}
+
+	// take over default transport to avoid redirect
+	tr := &http.Transport{}
```

Comment:
@BrianBland Yes it should use `dialTimeout`. Actually it should use previous `transport` variable directly.
I debug this before and use different variable with timeout to observe the performance.
They should be normalizing here. I would correct it.